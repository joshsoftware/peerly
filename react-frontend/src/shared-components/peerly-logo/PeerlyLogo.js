import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import PlusSign from "shared-components/plus-sign/PlusSign";

const Logo = styled.div`
  text-align: center;
  margin-top: 15px;
  width: 50px;
  height: 50px;
  border: 3px solid
    ${({ theme }) => (theme === "dark" ? "var(--white)" : "var(--atomic)")};
  background-color: ${({ theme }) =>
    theme === "dark" ? "var(--atomic)" : "var(--white)"};
  transform: rotate(45deg);
`;

const PeerlyLogo = ({ theme, fontSize }) => (
  <Logo theme={theme} className="mx-auto" data-testid="Logo">
    <PlusSign theme={theme} cross fontSize={fontSize} />
  </Logo>
);

PeerlyLogo.defaultProps = {
  theme: "dark",
  fontSize: "30px",
};

PeerlyLogo.propTypes = {
  theme: PropTypes.oneOf(["dark", "light"]),
  fontSize: PropTypes.string,
};

export default React.memo(PeerlyLogo);
