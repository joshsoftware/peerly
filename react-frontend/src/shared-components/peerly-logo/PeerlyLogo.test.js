import React from "react";
import { render } from "@testing-library/react";

import "setupTests";
import PeerlyLogo from "shared-components/peerly-logo/PeerlyLogo";

it("should render PeerlyLogo Component", () => {
  const { asFragment } = render(<PeerlyLogo theme="dark" fontSize="48px" />);
  expect(asFragment()).toMatchSnapshot();
});

test("should test color,fontsize and background of Peerly Logo", () => {
  const { getByTestId } = render(<PeerlyLogo theme="dark" />);
  expect(getByTestId("Logo")).toHaveStyleRule(
    "border",
    "3px solid var(--white)"
  );
  expect(getByTestId("Logo")).toHaveStyleRule(
    "background-color",
    "var(--atomic)"
  );
});
