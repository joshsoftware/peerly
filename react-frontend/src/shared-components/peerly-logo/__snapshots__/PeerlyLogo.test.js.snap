// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PeerlyLogo Component 1`] = `
<DocumentFragment>
  .c1 {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  font-size: 48px;
  color: var(--white);
}

.c0 {
  text-align: center;
  margin-top: 15px;
  width: 50px;
  height: 50px;
  border: 3px solid var(--white);
  background-color: var(--atomic);
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

<div
    class="c0 mx-auto"
    data-testid="Logo"
  >
    <div
      class="c1"
      data-testid="plusSignText"
      font-size="48px"
    >
      +
    </div>
  </div>
</DocumentFragment>
`;
